### PR TITLE
Add claims doc to msal-common

### PR DESCRIPTION
**Pull request by derisen**
Adds a document discussing client capabilities and claims challenges with reference to [ms.docs article](https://learn.microsoft.com/en-us/azure/active-directory/develop/claims-challenge) for more.

### CODE REVIEW

File: lib/msal-common/docs/claims.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+A resource server may deem an access token it receives as having insufficient claims. If the client application that sends the access token has declared itself to be capable, the resource server may respond with a claims challenge, upon which the client can acquire a new token satisfying the claims challenge.
+
+MSAL supports clients to declare themselves capable of claims challenges (i.e. `clientCapabilities`). This can be configured via the configuration object used during MSAL instantiation:
+
+- msal-browser [configuration](../../msal-browser/docs/configuration.md)
+- msal-node [configuration](../../msal-node/docs/configuration.md)
+
+MSAL's `login*` and `acquireToken*` APIs accepts a `claims` parameter as part of the [request object](https://azuread.github.io/microsoft-authentication-library-for-js/ref/modules/_azure_msal_common.html#baseauthrequest) to acquire a new token with the said claims.
+
+For more information, please refer to: [Claims challenges, claims requests, and client capabilities](https://learn.microsoft.com/en-us/azure/active-directory/develop/claims-challenge)
```
Comment (jasonnutter) [inline lib/msal-common/docs/claims.md]: Remove `en-us` from any links to docs.microsoft.com

**Comment by jasonnutter** [comment]
Should we also mention and links to docs for CAE? https://learn.microsoft.com/en-us/azure/active-directory/conditional-access/concept-continuous-access-evaluation

File: lib/msal-common/docs/claims.md (hunk 1 above)
Comment (derisen) [reply lib/msal-common/docs/claims.md]: oops -done

File: lib/msal-common/docs/claims.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Claims and Claims Challenges in MSAL
+
+A resource server may deem an access token it receives as having insufficient claims. If the client application that sends the access token has declared itself to be capable, the resource server may respond with a claims challenge, upon which the client can acquire a new token satisfying the claims challenge and try again.
```
Comment (bgavrilMS) [inline lib/msal-common/docs/claims.md]: I'm under the impression that the 2 concepts are orthogonal. CC @jmprieur

Comment (bgavrilMS) [reply lib/msal-common/docs/claims.md]: MSAL py docs seem more accurate to me: 

`Client capability is meant to inform the Microsoft identity platform (STS) what this client is capable for, so STS can decide to turn on certain features. For example, if client is capable to handle claims challenge, STS can then issue CAE access tokens to resources knowing when the resource emits claims challenge the client will be capable to handle.
`

I think AAD only supports "CP1" right now, which is what you described, but the mechanism is more general purpose.

Comment (derisen) [reply lib/msal-common/docs/claims.md]: Makes sense. I've break down the content into separate pages, with cross links to each other.

**Review by bgavrilMS — APPROVED** [review]

**Review by tnorling — APPROVED** [review]

**Review by hectormmg — APPROVED** [review]

**Pull request merged**